torch/amp/autocast_mode.py

```diff
@@ -197,6 +198,8 @@ def __init__(self, device_type : str,
             self.fast_dtype = torch.xpu.get_autocast_xpu_dtype()  # type: ignore[attr-defined]
         elif self.device == 'hpu':
             self.fast_dtype = torch.hpu.get_autocast_hpu_dtype()  # type: ignore[attr-defined]
+        elif self.device == self.custom_backend_name:
+            self.fast_dtype = eval('torch.{0}.get_autocast_{0}_dtype()'.format(self.custom_backend_name))
```

Comment:
hmmm so this is relying on the backend to run `torch._register_device_module('foo', FooModule)`, and for `FooModule` to have a method `get_autocast_foo_dtype()`?

Reply:
> hmmm so this is relying on the backend to run `torch._register_device_module('foo', FooModule)`, and for `FooModule` to have a method `get_autocast_foo_dtype()`?

yes,   it relys on the fun of  `torch._register_device_module`. As for the custom backend,  I have have analyzed several devices such as  intel XPU. Now, if anyone want to add new device to pytorch. Firsty, rename the backend with rename_privateuse1_backend, for example, the name is `foo`. And then add the ops and custom ops to the privateuse1 backend, it is the most important thing. As for new device, we have to subclass the c10::storageImpl if needed, allocator, and device management(stream, event), etc... in C++. And in Python, we need to export these related APIs. This APIs (without ops) in Python,  if we can call it with torch.foo.func, I think this will give users a good experience. So I have noticed that you have give a func to register a backend module with torch._register_device_module. So it is very good to use these two interfaces together, `torch._register_device_module` and `rename_privateuse1_backend`. And other optional features for new device such as AMP, the implement the interface (get_autocast_foo_dtype, etc...), could placed in the registered device module. But if the new device does not need the amp feature, just not using amp, it is ok. 
ha, if there is anything I understand wrong, please give me some advice.